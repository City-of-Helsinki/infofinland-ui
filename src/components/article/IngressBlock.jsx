import ParseHtml from '../ParseHtml'
import Block from '@/components/layout/Block'

const IngressBlock = ({field_description})=>(
<Block className="ifu-article__ingress text-body-large my-8">
  <ParseHtml html={field_description} />
</Block>)


export default IngressBlock
