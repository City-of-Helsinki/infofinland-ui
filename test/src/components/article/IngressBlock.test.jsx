/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import IngressBlock from '@/components/article/IngressBlock'

describe('IngressBlock', () => {
  it('should render the given HTML inside a Block', () => {
    const { asFragment } = render(
      <IngressBlock
        field_description={
          'Alkukartoituksessa arvioidaan, mitkÃ¤ palvelut auttavat sinua kotoutumaan.'
        }
      />
    )
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
