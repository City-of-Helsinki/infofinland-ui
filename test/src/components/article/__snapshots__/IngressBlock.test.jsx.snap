// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngressBlock should render the given HTML inside a Block 1`] = `
<DocumentFragment>
  <div
    class="px-6 lg:px-12 lg:mx-12  xl:mx-28 2xl:mx-48  3xl:ms-64  3xl:max-w-4xl my-16 text-body-large ifu-article__ingress"
  >
    Alkukartoituksessa arvioidaan, mitk√§ palvelut auttavat sinua kotoutumaan.
  </div>
</DocumentFragment>
`;
